### ISSUE TITLE
Support for nVidia tensorcore

### ISSUE THREAD

**Issue by rafale77** [davisking/dlib#2104]
More of a feature request than a problem report and forgive my ignorance if this is irrelevant but the nvidia 20x series and the 1660ti have tensor cores which could be use when called out on the nvidia driver using the fp16 extension. tensorflow does it that way. Is there a way to implement this on dlib?

**Comment by davisking** [comment]
Yeah, you can't use fp16 with dlib's dnn tooling right now.  It's all fp32.  If someone wants to go and update it to support fp16 that would be neat, although it's somewhat non-trivial.
